internal/rpcserver/rpcwebsocket.go

```diff
@@ -2173,6 +2261,81 @@ func handleStopNotifyNewTransactions(_ context.Context, wsc *wsClient, _ interfa
 	return nil, nil
 }
 
+// handleNotifyMixMessages implements the notifymixmessages command extension
+// for websocket connections.
+func handleNotifyMixMessages(_ context.Context, wsc *wsClient, _ interface{}) (interface{}, error) {
+	wsc.rpcServer.ntfnMgr.RegisterMixMessages(wsc)
+	return nil, nil
+}
+
+// handleStopNotifyMixMessages implements the stopnotifymixmessages command
+// extension for websocket connections.
+func handleStopNotifyMixMessages(_ context.Context, wsc *wsClient, _ interface{}) (interface{}, error) {
+	wsc.rpcServer.ntfnMgr.UnregisterMixMessages(wsc)
+	return nil, nil
+}
+
+func handleSendMixMessage(_ context.Context, wsc *wsClient, icmd interface{}) (interface{}, error) {
```

Comment:
Why is this only defined in `websocket.go`?  It should be on the regular RPC server and not require WebSockets.

It's also missing the map entries in `rpcHandlersBeforeInit` and `rpcLimited` assuming we want to allow limited (non-admin) RPC users to submit mix messages which I image.

Finally, the naming is inconsistent.  It's called `sendmixmessage` and some places and `submitmixmessage` in others.

I think the RPC should be called `sendrawmixmessage` to mirror `sendrawtransaction` for consistency.  Then, of course, this would be called `handleSendRawMixMessage` and so on.

Reply:
We are accessing wsc to send additional notifications, but i don't think this should be necessary.  taking a closer look at revamping this.

(there is also a long-standing issue where dcrd would always re-notify the messages we sent back to the same wallet.  we probably do want to notify any other websocket wallet who didn't send this message, but it is rather silly to notify the sender itself.)

Reply:
We can still pump out notifications without accessing this through the wsc, sorry i got confused by that type spaghetti.

And we still could i suppose, but i would like to revisit not notifying ourselves of our own submitted messages.  Maybe after release.